pkg/api/provision.go

```diff
@@ -411,13 +439,65 @@ func (s *server) provision(w http.ResponseWriter, r *http.Request) {
 		s.writeResponse(w, http.StatusInternalServerError, responseEmptyJSON)
 		return
 	}
-
 	// If we get all the way to here, we've been successful!
 	s.writeResponse(w, http.StatusAccepted, responseProvisioningAccepted)
 
 	log.WithFields(logFields).Debug("asynchronous provisioning initiated")
 }
 
+func (s *server) isParentProvisioning(instance service.Instance) (bool, error) {
+	//No parent, so no need to wait
+	if instance.ParentAlias == "" {
+		return false, nil
+	}
+
+	parent, parentFound, err := s.store.GetInstanceByAlias(instance.ParentAlias)
```

Comment:
Is this because `instance.Parent` will always be nil at this point in time?

Reply:
Yeah, at this point in time the instance.Parent is nil (we just created the instance) as we have not retrieved it from storage.  Just realized that should not need to do that in the tasks.

Reply:
I removed this from the task (but have not pushed).